src/components/inputs/text/NameTokenSelector/NameTokenSelector.tsx

```diff
@@ -30,177 +47,513 @@ function NameTokenSelector({
-      const txid = new ArweaveTransactionID(id);
-      selectedTokenCallback(txid);
-      const state =
-        await arweaveDataProvider.getContractState<PDNTContractJSON>(txid);
-      const contract = new PDNTContract(state, txid);
-
-      if (!contract.isValid()) {
-        throw Error('Contract does not match required schema');
+      if (!tokens) {
+        throw new Error('No Name Tokens Found');
       }
-      setSelectedNameToken({
-        id,
-        name: contract.name,
-        ticker: contract.ticker,
-      });
-      setValidImport(true);
-      setSearchText('');
+      const filteredResults = Object.keys(tokens)
+        .filter((id) => {
+          const { owner, controller, name, ticker } = tokens[id];
+          const queryResult = [owner, controller, name, ticker, id].some(
+            (term) =>
+              term &&
+              term.toString().toLowerCase().includes(query.toLowerCase()),
+          );
+          return queryResult;
+        })
+        .map((id) => {
+          const { name, ticker } = tokens[id];
+          return { id, name: name ?? '', ticker: ticker ?? '' };
+        })
+        .filter((n) => n !== undefined);
+      if (!filteredResults.length) {
+        throw new Error('No ANT tokens found for that search');
+      }
+
+      setFilteredTokens(filteredResults);
     } catch (error) {
+      setFilteredTokens(undefined);
       eventEmitter.emit('error', error);
-      setSelectedNameToken(undefined);
-      setValidImport(false);
     } finally {
       setSearching(false);
+      setListPage(1);
     }
   }
 
+  function handleSetToken({
+    id,
+    name,
+    ticker,
+  }: {
+    id?: string;
+    name?: string;
+    ticker?: string;
+  }) {
+    setSearchText('');
+    setFilteredTokens(undefined);
+
+    if (id === undefined) {
+      if (selectedToken === undefined) {
+        setSearchActive(false);
+      }
+      return;
+    }
+    setSelectedToken({ id, name: name ?? '', ticker: ticker ?? '' });
+    selectedTokenCallback(new ArweaveTransactionID(id));
+    setSearchActive(false);
+    setListPage(1);
+  }
+
+  const customPreviousAndNextButtons: PaginationProps['itemRender'] = (
+    page,
+    type,
+    originalElement,
+  ) => {
+    if (!tokens) {
+      return;
+    }
+    if (type === 'next' || type === 'prev') {
+      return;
+    }
+    if (type === 'page') {
+      return (
+        <span
+          className="flex flex-row hover center"
+          style={{
+            color: listPage == page ? 'white' : 'var(--text-faded)',
+            width: '32px',
+          }}
+        >
+          {page}
+        </span>
+      );
+    }
+    return originalElement;
+  };
+
+  function updatePage(page: number) {
+    setListPage(page);
+  }
+
   return (
-    <div
-      className="flex flex-column"
-      style={{
-        position: 'relative',
-        height: 'fit-content',
-        maxHeight: '400px',
-      }}
-    >
-      {/* input wrapper */}
+    <>
```

Comment:
extraneous js fragment

Reply:
fixed in https://github.com/ar-io/pdns-react/pull/229/commits/770ee8aa94514c233cac10ecee55f3a26abc37a7